### DISCUSSION TITLE
why not implement as wrapper?

### DISCUSSION THREAD

**Discussion by callowaylc** [mikefarah/yq#813] [General]
Just curious, but if the aim is to provide a "jq like" tool, that "doesn't yet support everything jq does", why not just convert between json and yaml and pass arguments directly to jq? 

```sh
# set first positional argument to "." 
$ set -- .
```
```sh
# create yaml, convert to json, run jq with positional arguments and convert back to yaml
$ printf "%s: %s\n" ba boo \
> | python -c 'import json, sys, yaml ; y=yaml.safe_load(sys.stdin.read()) ; print(json.dumps(y))' \
> | jq -re "$@" \
> | python -c 'import sys, yaml, json; j=json.loads(sys.stdin.read()); print(yaml.safe_dump(j))'
ba: boo

```

Achieves the same thing without the awful (not your code but text processing in general); not trying to be a dick, but have been curious about this one for awhile. 

*python one liner references*
https://gist.github.com/jaytaylor/f11428514269101049a225f20aea802f
https://gist.github.com/mboersma/1329671

**Comment by mikefarah** [comment]
Yeah that's definetly a valid approach, there are two reasons why you may not want to do that:

a) YAML is a super set of JSON - e.g. you can have multiple documents in a single file, there are comments, anchors and other things JSON doesn't have.

b) Depending on where you are processing the yaml - you may not want the overhead of installing python
